tests/Dfe.PlanTech.Web.UnitTests/TagHelpers/HeaderTagHelperTests.cs

```diff
@@ -11,6 +11,28 @@ namespace Dfe.PlanTech.Web.UnitTests.TagHelpers;
 
 public class HeaderComponentTagHelperTests
 {
+    private TagHelperContext CreateTagHelperContext()
+    {
+        return new TagHelperContext(
+            tagName: "header",
+            allAttributes: new TagHelperAttributeList(),
+            items: new Dictionary<object, object>(),
+            uniqueId: "header-test");
+    }
+
+    private TagHelperOutput CreateTagHelperOutput()
+    {
+        return new TagHelperOutput(
+            "header",
+            attributes: new TagHelperAttributeList(),
+            getChildContentAsync: (useCachedResult, encoder) =>
+            {
+                var tagHelperContent = new DefaultTagHelperContent();
+                tagHelperContent.SetContent("");
+                return Task.FromResult<TagHelperContent>(tagHelperContent);
+            });
+    }
+
     [Fact]
     public void Should_RenderH1Tag_When_H1()
```

Comment:
Pretty sure I wrote all these tests so don't shoot me, but whilst you're there it might be worth refactoring the tests into one? Ideally using `[Theory]` and `[InlineData]`. E.g.:

```cs
    [Theory]
    [InlineData(HeaderTag.H1, HeaderSize.Small)]
    [InlineData(HeaderTag.H1, HeaderSize.Medium)]
    //All the other combinations....
    public void Should_Render_When_Valid(HeaderTag headerTag, HeaderSize headerSize)
    {
        var header = new Header()
        {
            Text = "Header text",
            Tag = headerTag,
            Size = headerSize
        };

        //Rest of test....
    }
```

Or you could do something like this to save effort, and ensure that future additions to the tags + sizes are accommodated without having to change the tests:
```cs

    public void Should_Render_When_Valid()
    {
        var headerTags = Enum.GetValues<HeaderTag>();
        var headerSizes = Enum.GetValues<HeaderSize>();

        foreach (var tag in headerTags.Where(tag => tag != HeaderTag.Unknown))
        {
            foreach (var size in headerSizes.Where(size => size != HeaderSize.Unknown))
            {
                var header = new Header()
                {
                    Text = "Header text",
                    Tag = tag,
                    Size = size
                };

                //Rest of test
            }
        }

        //or to remove nested loops
        var combinations = headerTags.Where(tag => tag != HeaderTag.Unknown)
                                    .SelectMany(tag => headerSizes.Where(size => size != HeaderSize.Unknown), (tag, size) => new { Tag = tag, Size = size });

        foreach(var combination in combinations){
            //etc.
        }

        //or we can use actual LINQ, not just hte List extension methods
        var combinations = from tag in headerTags where tag != HeaderTag.Unknown
                       from size in headerSizes where where size != HeaderSize.Unknown
                       select new { Tag = tag, Size = size };

        foreach(var combination in combinations){
            //etc.
        }
    }
``` 

However the latter will fail on the first failure, whereas the former will essentially create a unit test per `InlineData`

YMMV

Reply:
Thanks, I've gone with option 1. Also spotted that we didn't have a test for Unknown HeaderSize so I've added one.